packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -112,8 +112,7 @@ abstract class TextSelectionControls {
   /// Subclasses can use this to decide if they should expose the paste
   /// functionality to the user.
   bool canPaste(TextSelectionDelegate delegate) {
-    // TODO(goderbauer): return false when clipboard is empty, https://github.com/flutter/flutter/issues/11254
-    return true;
+    return !Clipboard.isEmpty;
```

Comment:
If the clipboard was empty when the textfield was initiated, then copied something into the clipboard (outside of flutter), this will never become true right?

Reply:
I think we're adding some implementation knowledge leak here. This implies we know that showToolbar() first happens, where we cache things first, and then call buildToolbar on the TextSelectionControls subclass which then calls back canPaste (we don't necessarily know that) which then uses the cached data. This adds interpendencies and complexity and since TextSelectionControls.canPaste is a public API and we can't really control what users do with it, we can't let its output be only true conditionally. 

Unfortunately I don't have anything easier to suggest besides breaking the existing API and make TextSelectionControls.canPaste async. 

Then in _CupertinoTextSelectionControls and _MaterialTextSelectionControls, let it build a FutureBuilder based on super.canPaste and build a different _TextSelectionToolbar depending on the output. 

cc @Hixie, briefly chatted about this a while back. In case Ian has better proposals.